core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2942,13 +2936,13 @@ public static boolean isNumericType(Type type)
                 type instanceof DecimalType;
     }
 
-    private static boolean isExactNumericWithScaleZero(Type type)
+    private static boolean isNotExactNumericWithScaleZero(Type type)
```

Comment:
What's the benefit? Methods that perform negative checks are harder to reason and think about.

Reply:
every usage is negated and that seems weird to me but I won't fight over this :)